mne/decoding/time_gen.py

```diff
@@ -817,6 +817,172 @@ def _time_gen_one_fold(clf, X, y, train, test, scoring):
+        If None the classifier will be a standard pipeline including
+        StandardScaler and a linear SVM with default parameters.
+    train_times : dict | None
+        A dictionary to configure the training times:
+
+            ``slices`` : np.ndarray, shape (n_clfs,)
+                Array of time slices (in indices) used for each classifier.
+                If not given, computed from 'start', 'stop', 'length', 'step'.
+            ``start`` : float
+                Time at which to start decoding (in seconds). By default,
+                min(epochs.times).
+            ``stop`` : float
+                Maximal time at which to stop decoding (in seconds). By
+                default, max(times).
+            ``step`` : float
+                Duration separating the start of subsequent classifiers (in
+                seconds). By default, equals one time sample.
+            ``length`` : float
+                Duration of each classifier (in seconds). By default, equals
+                one time sample.
+
+        If None, empty dict. Defaults to None.
+    predict_mode : {'cross-validation', 'mean-prediction'}
+        Indicates how predictions are achieved with regards to the cross-
+        validation procedure:
+            ``cross-validation`` : estimates a single prediction per sample
+                based on the unique independent classifier fitted in the
+                cross-validation.
+            ``mean-prediction`` : estimates k predictions per sample, based on
+                each of the k-fold cross-validation classifiers, and average
+                these predictions into a single estimate per sample.
+
+        Default: 'cross-validation'
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+
+    Attributes
+    ----------
+    y_train_ : np.ndarray, shape (n_samples,)
+        The categories used for training.
+    estimators_ : list of list of sklearn.base.BaseEstimator subclasses.
+        The estimators for each time point and each fold.
+    y_pred_ : np.ndarray, shape (n_train_times, n_test_times, n_epochs,
+                                 n_prediction_dims)
+        Class labels for samples in X.
+    scores_ : list of lists of float
+        The scores (mean accuracy of self.predict(X) wrt. y.).
+        It's not an array as the testing times per training time
+        need not be regular.
+    test_times_ : dict
+        The same structure as ``train_times``.
+    cv_ : CrossValidation object
+        The actual CrossValidation input depending on y.
+
+    Notes
+    -----
+    The function implements the method used in:
+
+        Jean-Remi King, Alexandre Gramfort, Aaron Schurger, Lionel Naccache
+        and Stanislas Dehaene, "Two distinct dynamic modes subtend the
+        detection of unexpected sounds", PLOS ONE, 2013
+
+    .. versionadded:: 0.9.0
+
+    Returns
+    -------
+    """
+
+    def predict(self, X, test_times='diagonal', **kwargs):
```

Comment:
out of curiousity, this does not compute the entire GAT matrix and return the diagonal but rather computes just the diagonal, right?